### PR TITLE
Label some code referenced by main

### PR DESCRIPTION
**Pull request by meithecatte**

### CODE REVIEW

File: include/functypes/funcptr_02017798.h [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #ifndef POKEPLATINUM_FUNCPTR_02017798_H
 #define POKEPLATINUM_FUNCPTR_02017798_H
 
-typedef void (* UnkFuncPtr_02017798)(void *);
+typedef void (* Callback)(void *);
```
Comment (lhearachel) [inline include/functypes/funcptr_02017798.h:4]: Any chance we could move this into an appropriate header based on the usages?

```
$ grep -r 'functypes/funcptr_02017798.h' include/
include/core_sys.h:#include "functypes/funcptr_02017798.h"
include/unk_02017728.h:#include "functypes/funcptr_02017798.h"

$ grep -r 'functypes/funcptr_02017798.h' src/
src/unk_02017728.c:#include "functypes/funcptr_02017798.h"
```

File: src/main.c [hunk 2]
```diff
@@ -113,53 +115,53 @@ void NitroMain (void)
...
+            CheckHeapCanary();
             sub_02000E54();
             sub_0201CDD4(gCoreSys.unk_18);
             sub_0201CDD4(gCoreSys.unk_24);
 
             if (!gCoreSys.unk_30) {
                 OS_WaitIrq(1, OS_IE_V_BLANK);
-                gCoreSys.unk_2C++;
+                gCoreSys.frameCounter++;
             }
```
Comment (lhearachel) [inline src/main.c:144]: I'm wondering if this is the 30-FPS lock that the gen4 games are infamous for. That `OS_WaitIrq` call looks suspicious.

File: include/core_sys.h [hunk 3]
```diff
@@ -13,18 +13,18 @@ enum ButtonMode {
...
 
 typedef struct CoreSys {
-    UnkFuncPtr_02017798 unk_00;
-    void * unk_04;
-    UnkFuncPtr_02017798 unk_08;
-    void * unk_0C;
+    Callback mainCallback;
+    void * mainCallbackData;
+    Callback hblankCallback;
+    void * hblankCallbackData;
```
Comment (lhearachel) [inline include/core_sys.h:19]: Pull the names on these pointer-vars back a character.

```suggestion
    Callback mainCallback;
    void *mainCallbackData;
    Callback hblankCallback;
    void *hblankCallbackData;
```

File: include/core_sys.h [hunk 4]
```diff
@@ -43,11 +43,11 @@ typedef struct CoreSys {
...
     u8 unk_65;
     u8 unk_66;
-    u8 unk_67;
-    u8 unk_68;
+    u8 inhibitSleep;
+    u8 inhibitReset;
     u8 padding_69[3];
     BOOL unk_6C;
-    u32 * unk_70;
+    u32 * heapCanary;
```
Comment (lhearachel) [inline include/core_sys.h:50]: Here, as well, as above.

```suggestion
    u32 *heapCanary;
```

File: include/unk_02017728.h [hunk 5]
```diff
@@ -4,24 +4,24 @@
 #include "functypes/funcptr_02017798.h"
 
 void sub_02017728(void);
-void sub_02017798(UnkFuncPtr_02017798 param0, void * param1);
-void sub_020177A4(void);
-BOOL sub_020177BC(UnkFuncPtr_02017798 param0, void * param1);
+void SetMainCallback(Callback cb, void * data);
+void DisableHBlank(void);
+BOOL SetHBlankCallback(Callback cb, void * data);
```
Comment (lhearachel) [inline include/unk_02017728.h:9]: Here, as above.

```suggestion
void SetMainCallback(Callback cb, void *data);
void DisableHBlank(void);
BOOL SetHBlankCallback(Callback cb, void *data);
```

**Review by lhearachel — CHANGES_REQUESTED** [review]

**Comment by meithecatte** [comment]
Thank you for the suggestions. I applied them and rebased onto current `main`.

**Review by lhearachel — APPROVED** [review]

**Pull request merged**